src/components/RocketsDependencies/ReserveButtonComponent.js

```diff
@@ -0,0 +1,37 @@
+import React from "react";
+import PropTypes from 'prop-types';
+import { Button }  from "react-bootstrap";
+
+
+const ReserveButton = ({ onClickEvent, reserved, rocketID }) => {
+  let text; let variant;
+  if (reserved){
+    text = "Cancel Reservation"
+    variant = "secondary"
+  } else {
+    text = "Reserve"
+    variant = "primary"
+  }
+  return (
+    <Button
+      variant={variant}
+      onClick={() => onClickEvent(rocketID)}
```

Comment:
- [ ] [OPTIONAL] I would recommend directly dispatching a redux action instead of calling a function that was received as a prop, which only dispatches an action in the parent component. This would save you from passing one prop from the parent component and would make your code more readable.